File: back-office/client-side/src/model/V_payroll.ts
export default interface V_payroll {
  date?: Date;
  number?: number;
  matricule?: number;
  cnaps?: number;
  //   id: number;
  fullName: string;
  dtembauche?: Date;
  month_abs?: Date;
  classification?: string;
  fonction: string;
  service: string;
  poste: string;
  startingSalary: number;
  monthBaseSalary?: number;
  indemnite?: number;
  grossSalary?: number;
  cnaps_1?: number;
  ostie_1?: number;
  ostie_5?: number;
  netSalary?: number;
  netToPay?: number;
  netOfMonth?: number;
}

File: spring-server/src/main/java/com/example/springserve/v_payroll/V_payroll.java
package com.example.springserve.v_payroll;

import java.sql.Date;

import org.hibernate.annotations.Immutable;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

@Entity(name = "v_payroll")
@Immutable
public class V_payroll {

    @Id
    @Column(name = "id")
    public Long id;

    @Column(name = "full_name")
    public String fullName;

    @Column(name = "dtembauche")
    public Date dtembauche;

    @Column(name = "fonction")
    public String fonction;

    @Column(name = "service")
    public String service;

    @Column(name = "poste")
    public String poste;

    @Column(name = "starting_salary")
    public Double startingSalary;
}
